mllib-tests/v1p5/src/main/scala/mllib/perf/MLAlgorithmTests.scala

```diff
@@ -591,28 +623,72 @@ class DecisionTreeTest(sc: SparkContext) extends DecisionTreeTests(sc) {
     if (labelType == 0) {
       // Regression
       ensembleType match {
-        case "RandomForest" =>
-          Left(RandomForest.trainRegressor(rdd, categoricalFeaturesInfo, numTrees, featureSubsetStrategy,
-            "variance", treeDepth, maxBins, this.getRandomSeed))
-        case "GradientBoostedTrees" =>
+        case "mllib.RandomForest" =>
+          MLlibRFModel(RandomForest.trainRegressor(rdd, categoricalFeaturesInfo, numTrees,
+            featureSubsetStrategy, "variance", treeDepth, maxBins, this.getRandomSeed))
+        case "ml.RandomForest" =>
+          val sqlContext = new SQLContext(rdd.sparkContext)
+          val dataset = sqlContext.createDataFrame(rdd)
+          val model = new RandomForestRegressor()
+            .setImpurity("variance")
+            .setMaxDepth(treeDepth)
+            .setMaxBins(maxBins)
+            .setNumTrees(numTrees)
+            .setFeatureSubsetStrategy(featureSubsetStrategy)
+            .setSeed(this.getRandomSeed)
+            .fit(dataset)
+          MLRFRegressionModel(model)
+        case "mllib.GradientBoostedTrees" =>
           val treeStrategy = new Strategy(Algo.Regression, Variance, treeDepth,
             labelType, maxBins, QuantileStrategy.Sort, categoricalFeaturesInfo)
-          val boostingStrategy = BoostingStrategy(treeStrategy, SquaredError, numIterations = 100,
+          val boostingStrategy = BoostingStrategy(treeStrategy, SquaredError, numTrees,
             learningRate = 0.1)
-          Right(GradientBoostedTrees.train(rdd, boostingStrategy))
+          MLlibGBTModel(GradientBoostedTrees.train(rdd, boostingStrategy))
+        case "ml.GradientBoostedTrees" =>
+          val sqlContext = new SQLContext(rdd.sparkContext)
+          val dataset = sqlContext.createDataFrame(rdd)
+          val model = new GBTRegressor()
```

Comment:
OK, now regression compares mllib GBTs with L2 loss to ml GBTs with logistic loss. Is this inconsistency ok, or should we switch mllib GBT regression to also use logistic loss (since that's the only loss type supported by ML GBTs atm)?